src/connector/src/parser/mod.rs

```diff
@@ -390,15 +437,63 @@ async fn into_chunk_stream<P: ByteStreamSourceParser>(mut parser: P, data_stream
                             }
                         }
                     };
-                    builder.row_writer().fulfill_meta_column(f)?;
+                            builder.row_writer().fulfill_meta_column(f)?;
+                        }
+                    }
+                }
+
+                Ok(Either::Right(txn_ctl)) => {
+                    match txn_ctl {
+                        TransactionControl::Begin { id } => {
+                            if let Some(current_id) = &current_transaction {
+                                tracing::warn!(
+                                    "already in transaction, current: {}, new: {}",
+                                    current_id,
+                                    id
+                                );
+                            }
+                            current_transaction = Some(id);
+                        }
+                        TransactionControl::Commit { id } => {
+                            if current_transaction.as_ref() != Some(&id) {
+                                tracing::warn!(
+                                    "transaction id mismatch, current: {:?}, actual: {}",
+                                    current_transaction,
+                                    id
+                                );
+                            }
+                            current_transaction = None;
+                        }
+                    }
+
+                    // Not in a transaction anymore and `yield_asap` is set, so we should yield the
+                    // chunk now.
+                    if current_transaction.is_none() && yield_asap {
+                        yield_asap = false;
+                        yield StreamChunkWithState {
+                            chunk: builder.take(batch_len - (i + 1)),
+                            split_offset_mapping: Some(std::mem::take(&mut split_offset_mapping)),
+                        };
+                    }
+                }
+
+                Err(e) => {
+                    tracing::warn!("message parsing failed {}, skipping", e.to_string());
+                    // This will throw an error for batch
+                    parser.source_ctx().report_user_source_error(e)?;
+                    continue;
                 }
             }
         }
 
-        yield StreamChunkWithState {
-            chunk: builder.finish(),
-            split_offset_mapping: Some(split_offset_mapping),
-        };
+        // If we are not in a transaction, we should yield the chunk now.
+        if current_transaction.is_none() {
```

Comment:
Shall we limit the max size of a transaction? Similar with the limitation for DML transaction. Otherwise, it's possible that a very large transaction from upstream DB causes RW to crash
 
https://github.com/risingwavelabs/risingwave/blob/0a9d2b44f52e7540120f5ba945c73863b50aabc9/src/stream/src/executor/dml.rs#L57-L64

Reply:
Done. PTAL 🥰

Reply:
I think we need to unify these 2 numbers and document it somewhere in our docs. cc. @chenzl25

Reply:
+1. We need to document it in DML and table CDC section.

Reply:
@chenzl25 @fuyufjh @BugenZhao Please see [this doc PR](https://github.com/risingwavelabs/risingwave-docs/pull/1279) for the related doc updates. Please review and make edits as you see fit. Thanks.